gammapy/scripts/make_bg_cube_models.py

```diff
@@ -0,0 +1,329 @@
+        print(repr(azimuth_edges))
+
+    # wrap azimuth angles to (-90, 270) deg
+    # TODO: needs re-thinking if azimuth angle definitions change!!!
+    #       or if user-defined azimuth angle bins are allowed!!!
+    azimuth = Angle(observation_table['AZ_PNT']).wrap_at(Angle(270., 'degree'))
+    observation_table['AZ_PNT'] = azimuth
+
+    # create output folder if not existing
+    outdir = os.environ['PWD'] + '/splitted_obs_list/'
+    if not os.path.isdir(outdir):
+        os.mkdir(outdir)
+    else:
+        # clean folder if available
+        for oldfile in os.listdir(outdir):
+            os.remove(outdir + oldfile)
+
+    # loop over altitude and azimuth angle bins: remember 1 bin less than bin boundaries
+    for i_alt in range(len(altitude_edges) - 1):
+        if DEBUG:
+            print()
+            print("bin alt", i_alt)
+        for i_az in range(len(azimuth_edges) - 1):
+            if DEBUG:
+                print()
+                print("bin az", i_az)
+
+            # filter observation table
+            observation_table_filtered = observation_table
+
+            selection = dict(type='par_box', variable='ALT_PNT',
+                             value_range=(altitude_edges[i_alt], altitude_edges[i_alt + 1]))
+            observation_table_filtered = observation_table_filtered.select_observations(selection)
+
+            selection = dict(type='par_box', variable='AZ_PNT',
+                             value_range=(azimuth_edges[i_az], azimuth_edges[i_az + 1]))
+            observation_table_filtered = observation_table_filtered.select_observations(selection)
+
+            if DEBUG:
+                print(observation_table_filtered)
+
+            # skip bins with no obs
+            if len(observation_table_filtered) == 0:
+                continue # skip the rest
+
+            # save the observation list to a fits file
+            outfile = outdir +\
+                     'bg_observation_table_alt{0}_az{1}.fits.gz'.format(i_alt, i_az)
+            if DEBUG:
+                print("outfile", outfile)
+            observation_table_filtered.write(outfile)
+
+
+def stack_observations(fits_path):
+    """Stack events for each observation group (bin) and make background model.
+
+    The models are stored into FITS files.
+
+    Parameters
+    ----------
+    fits_path : str
+        path to dir containing list of input fits event files
+    """
+    if DEBUG:
+        print()
+        print("###############################")
+        print("# Starting stack_observations #")
+        print("###############################")
+
+    # create output folder if not existing
+    outdir = os.environ['PWD'] + '/bg_cube_models/'
+    if not os.path.isdir(outdir):
+        os.mkdir(outdir)
+    else:
+        # clean folder if available
+        for oldfile in os.listdir(outdir):
+            os.remove(outdir + oldfile)
+
+    # loop over altitude and azimuth angle bins: remember 1 bin less than bin boundaries
```

Comment:
Again, this function only works for exactly this grouping in `(azimuth, zenith)`.
It would be much more flexible and better to have one for loop over observation groups here, i.e. to have the observation grouping and background modeling decoupled!